dao/src/main/java/com/fitgoal/dao/impl/AuditDaoImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.fitgoal.dao.impl;
+
+import com.fitgoal.dao.AuditDao;
+import com.fitgoal.dao.domain.AuditDto;
+import com.fitgoal.dao.util.AuditMongoConverter;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import org.bson.Document;
+import lombok.NonNull;
+import org.bson.types.ObjectId;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static com.fitgoal.dao.util.AuditDtoFields.*;
+import static com.mongodb.client.model.Filters.eq;
+
+public class AuditDaoImpl implements AuditDao {
+
+    private static final String COLLECTION_NAME = "audit";
+    private static final String DATABASE = "notification_db";
+
+
+    private final MongoCollection<Document> collection;
+
+    @Inject
+    public AuditDaoImpl(@NonNull MongoClient mongo) {
+        this.collection = mongo.getDatabase(DATABASE).getCollection(COLLECTION_NAME);
+    }
+
+    @Override
+    public void create(AuditDto auditDto) {
+        collection.insertOne(AuditMongoConverter.auditDtoToDocument(auditDto));
+    }
+
+    @Override
+    public Optional<AuditDto> findById(String id) {
+        Document document = collection.find(eq(ID, new ObjectId(id))).first();
+
+        return Optional.ofNullable(document)
+                .map(AuditMongoConverter::documentToAuditDto);
+    }
+
+    @Override
+    public List<AuditDto> findAll() {
+        Spliterator<Document> mongoCursor = collection.find().spliterator();
+        return StreamSupport.stream(mongoCursor, false) .map(AuditMongoConverter::documentToAuditDto) .collect(Collectors.toList());
+    }
+
+    @Override
+    public AuditDto update(AuditDto auditDto) {
+        Document id = new Document(ID, auditDto.getId());
+        Document updatedDocument =  new Document("$set", AuditMongoConverter.auditDtoToDocument(auditDto));
```

Comment:
Are you sure that updating the entire document works this way?

Reply:
Mistake fixed. Thanks!
```
   @Override
    public AuditDto update(AuditDto auditDto) {
        Document id = new Document(ID, new ObjectId(auditDto.getId()));
        Document updatedDocument = new Document("$set", AuditMongoConverter.auditDtoToDocument(auditDto, auditDto.getId()));
        collection.updateOne(id, updatedDocument);
        return auditDto;
    }
```